internal/handlers/handler_user_info.go

```diff
@@ -19,6 +18,13 @@ func UserInfoGet(ctx *middlewares.AutheliaCtx) {
 		return
 	}
 
+	if changed := userInfo.SetDefaultMethod(ctx.AvailableSecondFactorMethods()); changed {
```

Comment:
the method name could be clearer. Usually the "set" prefix is for setters, i.e., dumb methods setting an attribute of the object but in this case, the method seems to be more complex according to the signature returning a bool so I suggest renaming it to ChangeDefaultMethod instead.

Reply:
However, taking a step back, I think updating objects in a GET HTTP endpoint is not a good idea in general in terms of the user contract. It violates the semantic of GET.
IMO, to workaround this issue, this corner case should be treated in the client (browser) instead. The client should retrieve the available factors and compare to the retrieved user method. If, still in the client, the user method is not available then the client should send a POST to set the new method derived from one available.